build-ci/github_protect_travis_flake8.rst

```diff
@@ -0,0 +1,79 @@
+#############################################
+Github Branch Protection, TravisCI and Flake8
```

Comment:
There are distinct items here. Github configurations are not related to TravisCI jobs. I would expect "how to configure your github repo" to be in a different place in the developer guide. It should describe the recommended way of setting up a repository and provide the command line to automate that. Then you have a page describing what Travis can do and then informing the user how to enable specific options. Flake8 then becomes an example. Checking the build of latex/sphinx documents becomes another.